include/device.h

```diff
@@ -877,9 +882,11 @@ __deprecated static inline int device_usable_check(const struct device *dev)
  */
 BUILD_ASSERT(sizeof(device_handle_t) == 2, "fix the linker scripts");
 #define Z_DEVICE_DEFINE_HANDLES(node_id, dev_name, ...)			\
-	extern const device_handle_t					\
+	extern COND_CODE_0(CONFIG_HAS_DYNAMIC_DEVICES,			\
+		    (const device_handle_t), (device_handle_t))		\
 		Z_DEVICE_HANDLE_NAME(node_id, dev_name)[];		\
-	const device_handle_t						\
+	COND_CODE_0(CONFIG_HAS_DYNAMIC_DEVICES,				\
+		    (const device_handle_t), (device_handle_t))		\
```

Comment:
note: the only thing being modified here is the `const` attribute for device handles, so I think a better option would be something like:

```
#ifdef CONFIG_DEVICE_HANDLES_ROMABLE
#define Z_DEVICE_HANDLE_CONST const
#else
#define Z_DEVICE_HANDLE_CONST
#endif
```

and make `CONFIG_DEVICE_HANDLES_ROMABLE` a hidden option selected by PM domains (name may be changed, it's the first that came to my mind). Then, just use `Z_DEVICE_HANDLE_CONST` instead of `const` everywhere.

p.d. I did something similar for pinctrl https://github.com/zephyrproject-rtos/zephyr/blob/main/include/drivers/pinctrl.h#L194-L204

Reply:
I liked the idea, just the new Kconfig symbol that I don't think it is necessary. We already have the HAS_DYNAMIC_DEVICE or whatever other name we come up. Basically,
```
#ifdef CONFIG_HAS_DYNAMIC_DEVICES
#define Z_DEVICE_HANDLE_CONST const
#else
#define Z_DEVICE_HANDLE_CONST
#endif
```

Reply:
I'm still confused on what a "dynamic device" is

Reply:
forgot the name, it can be changed. I just mentioned that I don't want to introduce a redundant symbol only for the constant attribute. We should be able to re-use the same symbol everywhere, `HANDLES_ROMABLE` does not say anything about why we have this condition as well. These hidden symbols are pretty hard to maintain and also make the code more difficult to maintain. One hidden symbol for this feature is enough.